code/modules/admin/verbs/adminhelp.dm

```diff
@@ -210,6 +210,15 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
  * * is_bwoink - Boolean operator, TRUE if this ticket was started by an admin PM
  */
 /datum/admin_help/New(msg_raw, client/C, is_bwoink, urgent = FALSE)
+	//check message length
+	var/message_length = length(msg_raw)
+	if(message_length > MAX_MESSAGE_LEN)
+		if(C)
+			to_chat(C, span_redtext("Failed to send your adminhelp. Your message is [message_length] characters which exceeds the maximum length of [MAX_MESSAGE_LEN], please reduce the length and try again. You attempted to send the following message:\n[sanitize(msg_raw)]"))
```

Comment:
I'm unconvinced this is the right approach to preserving the message. Chances are it's vexingly long and going to distract or confuse the player away from the actual warning when done this way. If anything, the panel should be reopened, with the message itself, in its integrity, as default input.

Also, maybe I'm petty, but to be precise, this is byte-length, and non-ASCII characters take more of them.

```suggestion
			to_chat(C, span_redtext("Failed to send your adminhelp. Your message is [message_length] [span_tooltip("bytes to be precise. non-ASCII characters take more space", "characters")] which exceeds the maximum length of [MAX_MESSAGE_LEN], please reduce the length and try again."))
```